### PR TITLE
:lipstick: DOP-4635 applies styles in footnote to support dark mode

### PR DESCRIPTION
**Pull request by caesarbell**
### Stories/Links:

DOP-4635

### Current Behavior:

[Docs Manua: Footnote Oracle Linuxl](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-enterprise-on-red-hat-tarball/#footnote-oracle-linux)

### Staging Links:

[Staged: Docs Manua: Footnote Oracle Linuxl](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/docs/caesarbell/DOP-4635/tutorial/install-mongodb-enterprise-on-red-hat-tarball/index.html#footnote-oracle-linux)

### Notes:

- Applies styles for `:target` and the anchor tags inside the component. The styles should only take effect within this component and should not bleed into other anchor tags.
- For more context, feedback from the designers can be found [here](https://mongodb.slack.com/archives/C076Z47AFTL/p1719248265218619)

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Review by mmeigs — COMMENTED** [review]
Chill! 
There's one aberration from prod. The footnote anchor tags in prod (light mode) are blue. And now in staging (light mode) they are black-ish. 

And about the same anchor tags in dark mode, could we check with design? I'm sure this is the color they gave you, but can we show them a pic of dark mode? I feel like the contrast between blue.light1 and the gray background are not accessible.

**Comment by mayaraman19** [comment]
overall it looks good. not sure if this is part of this specific component/ticket, but if a footnote has already been clicked (at least on my computer) the resulting link is quite difficult to see in dark mode. just want to note this somewhere in the case that it's not part of this ticket.
image

**Comment by caesarbell** [comment]
> FootnoteReference

@mayaraman19 this is an interesting question. I wonder if there is another ticket for the `FootnoteReference` component. The ticket for this work only focused on the `Footnote` component, and it seems like previously they have tried to keep one component per ticket. 

I can see if there is a ticket for the `FootnoteReference` and if not make one or ask for it to be made. Unless the team feels I should add the work to this ticket/PR.

**Review by mayaraman19 — APPROVED** [review]
generally looks good, but wondering if we want to implement the removal of `style` here talked about offline? although,  I'm not 100% sure if we're waiting on anything to start that work and don't mind if we push it to a later ticket.

**Comment by caesarbell** [comment]
> generally looks good, but wondering if we want to implement the removal of `style` here talked about offline? although, I'm not 100% sure if we're waiting on anything to start that work and don't mind if we push it to a later ticket.

From what I remember we are holding off until we figure out the priority of the component, which I think we will start with the anchor tag. With that said newer tickets I will probably start implementing the newer strategy. 🤷

File: src/components/Footnote/index.js [hunk 1]
```diff
@@ -38,8 +48,26 @@ const Footnote = ({ nodeData: { children, id, name }, ...rest }) => {
...
+      '--background-color': palette.gray.dark2,
+      '--color': palette.blue.light1,
+    };
+  }
   return (
-    <table className="header-buffer" css={tableStyling} frame="void" id={`footnote-${ref}`} rules="none">
+    <table
+      className="header-buffer"
+      css={tableStyling(darkMode)}
+      style={footnoteDynamicStyles}
```
Comment (seungpark) [inline src/components/Footnote/index.js:66]: i thought we were not doing style attributes anymore as mentioned by @mayaraman19 ? would prefer the className styling to not interfere with smartling translation moving forward

Comment (caesarbell) [reply src/components/Footnote/index.js:66]: @seungpark I thought we decided to hold off until we got the report from @rayangler on which component causes the most problems. But maybe I misunderstood. I can have a follow-up PR for this change.

Comment (rayangler) [reply src/components/Footnote/index.js:66]: I think footnotes are pretty uncommon and low enough risk that this is okay for now. Let's see what Smartling says about how the Link component change after the next release, and then I can take it upon myself to conform everything together if needed.

**Pull request merged**